src/modules/courses/complete/complete.service.ts

```diff
@@ -56,15 +58,42 @@ export class CompleteService {
 			.findOneBy({ userId, courseId });
 		if (enrollment.completed === true) return;
 
-		const { affected } = await this.dataSource
-			.getRepository(EnrollmentEntity)
-			.update(
-				{ userId, courseId },
-				{ completed: true, completedAt: new Date() },
-			);
+		const course = enrollment.courseId;
+		const wholeLectures = await this.dataSource
+			.getRepository(LectureEntity)
+			.find({
+				where: { courseId: course },
+			});
+		const finishedLectures = await this.dataSource
+			.getRepository(History)
+			.find({
+				where: {
+					userId,
+					isFinished: true,
+					lecture: {
+						courseId: course,
+					},
+				},
+				relations: ['lecture'],
+				select: {
+					lecture: {
+						courseId: true,
+					},
+				},
+			});
+
+		if (wholeLectures.length === finishedLectures.length) {
+			const { affected } = await this.dataSource
+				.getRepository(EnrollmentEntity)
+				.update(
+					{ userId, courseId },
+					{ completed: true, completedAt: new Date() },
+				);
 
-		if (affected !== 1) throw new InternalServerErrorException();
+			if (affected !== 1) throw new InternalServerErrorException();
 
-		return status(201);
+			return status(201);
+		}
+		return status(400, 'No all lectures are finished yet.');
```

Comment:
```suggestion
		//return status(400, 'No all lectures are finished yet.');
		return status(201);
```
- 201을 그냥 줘도 큰 문제가 없을것 같다고 생각이 되는데요. 혹시 어떠신가요??

Reply:
조금 애매하기는 하네요.
완강을 하지 않은 경우 생성되거나 변경된 리소스가 없음에도 201(Created)를 반환하는게 맞나 싶은 생각이 들었어요.
400 코드도 규민님 설명을 들어보니 적절하진 않은 것 같고, 200을 반환하는건 어떨까요?

Reply:
편하신대로 해주시면 될 것 같아요. 에러로 처리만 하지 않으면 모두 생각의 영역이라..

이 api 자체가 특정 강의가 완료되었으니 업데이트 해주세요! 느낌이라서, 그 부분만 업뎃 잘 됐으면 상관 없을것 같습니다.